src/Symfony/Component/RateLimiter/RateLimit.php

```diff
@@ -67,7 +68,13 @@ public function getLimit(): int
 
     public function wait(): void
     {
-        $delta = $this->retryAfter->format('U.u') - microtime(true);
+        if (null !== $this->clock) {
+            $now = $this->clock->now()->format('U.u');
```

Comment:
if the only method you rely on is `->now()`, please use the PSR ClockInterface as typehint instead of the Symfony one, to make the code more reusable.